.github/workflows/release.yml

```diff
@@ -0,0 +1,606 @@
+        --cross-file misc/ci/emscripten-ephemeral-ci.ini
+        --cross-file misc/ci/emscripten-build-test-ci.ini
+        --prefix=$(pwd)/build-release
+
+        meson configure
+        -Dbuild.cpp_std=gnu++14
+        build/emscripten
+
+    - name: Build
+      run: |
+        source ${{ env.EM_CACHE_FOLDER }}/emsdk_env.sh
+        # TODO: fix race condition in SPIRV-Tools headers
+        meson compile -C build/emscripten --verbose || true
+        meson compile -C build/emscripten --verbose
+
+    - name: Package
+      run: |
+        source ${{ env.EM_CACHE_FOLDER }}/emsdk_env.sh
+        ninja txz -C build/emscripten --verbose
+
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v2
+      if: success()
+      with:
+        name: emscripten-wasm32-${{ steps.vars.outputs.tag }}-txz
+        path: build/emscripten/Taisei-${{ steps.vars.outputs.tag }}-emscripten-wasm32.tar.xz
+        if-no-files-found: error
+
+    - name: Upload Log
+      uses: actions/upload-artifact@v2
+      with:
+        name: taisei_emscripten_build_log
+        path: build/emscripten/meson-logs/meson-log.txt
+        if-no-files-found: warn
+
+  switch-release-build:
+    name: Switch (ARM64)
+    runs-on: ubuntu-20.04
+    container: taiseiproject/switch-toolkit:20210520
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+        fetch-depth: 0
+
+    - name: Set Build Env
+      id: vars
+      run: |
+        git fetch --force --tags
+        echo ::set-output name=tag::${GITHUB_REF#refs/tags/v}
+        switch/crossfile.sh > misc/ci/switch-crossfile-ci.ini
+
+    - name: Configure
+      run: >
+        meson setup build/nx
+        --cross-file=misc/ci/switch-crossfile-ci.ini
+        --prefix=$(pwd)/build-release
+
+    - name: Build
+      run: meson compile -C build/nx --verbose
+
+    - name: Package
+      run: ninja zip -C build/nx --verbose
+
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v2
+      if: success()
+      with:
+        name: switch-aarch64-${{ steps.vars.outputs.tag }}-zip
+        path: build/nx/Taisei-${{ steps.vars.outputs.tag }}-switch-aarch64.zip
+        if-no-files-found: error
+
+    - name: Upload Log
+      uses: actions/upload-artifact@v2
+      with:
+        name: taisei_switch_release_log
+        path: build/nx/meson-logs/meson-log.txt
+        if-no-files-found: warn
+
+  source-release:
+    name: Source (Archive)
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - name: Set Build Env
+      id: vars
+      run: |
+        git fetch --force --tags
+        echo ::set-output name=tag::${GITHUB_REF#refs/tags/v}
+
+    - name: Compress Source
+      run: >
+        tar --exclude-vcs -cf - ./ |
+        xz -9e --threads=8 -c - >
+        Taisei-${{ steps.vars.outputs.tag }}-source.tar.xz
```

Comment:
We rely on `meson dist` to package sources. It does a bit more than this, including running custom dist scripts registered from the build system (which we currently do not use, but we might eventually). Unfortunately it requires a working build directory, and it also does a test build with a clean source dir and default options as a sanity check (this can be disabled with `--no-tests`). For practical purposes, we can probably run it as a step in the Linux builder instead of its own container.

The naming scheme of the source archives is also different — `taisei-v{{version}}.tar.xz`. Unfortunately inconsistent with binary packages, but I'd prefer not to change it for now, to minimize friction for package maintainers and whoever else might rely on our URLs.

Reply:
Will do.